documentation/sdkautomation/sdk_customization.md

```diff
@@ -0,0 +1,762 @@
+          "mainRepository": "Azure/azure-sdk-for-python-pr"
+        }
+      }
+    }
+  }
+}
+```
+
+#### SpecRepoConfig Schema
+``` jsonc
+{
+  "type": "object",
+  "properties": {
+    "sdkRepositoryMappings": {
+      // A mapping of SDK repository names to the names of the SDK repositories
+      // that all interaction should go to instead.
+      "type": "object",
+      "additionalProperties": {
+        "$ref": "#/definitions/SdkRepositoryConfig"
+      },
+      "propertyNames": {
+        // The property name is the sdk name identifier.
+        "type": "string"
+      }
+    },
+    "overrides": {
+      // Override config for specific repository.
+      "type": "object",
+      "additionalProperties": {
+        "$ref": "#/"
+      },
+      "propertyNames": {
+        // The property name is the sdk repo ref.
+        "$ref": "#/definitions/RepositoryName"
+      }
+    },
+    "required": [
+      "sdkRepositoryMappings"
+    ]
+  },
+  "definitions": {
+    "RepositoryName": {
+      // Reference to a repository on github. Could be <repo> or <owner>/<repo>.
+      // By default the <owner> is the same as the owner of the spec repo.
+      "type": "string"
+    },
+    "SdkRepositoryConfig": {
+      "type": "object",
+      "properties": {
+        "mainRepository": {
+          // The repository that the final release PR will targeting.
+          "$ref": "#/definitions/RepositoryName"
+        },
+        "mainBranch": {
+          // Base branch of codegen branches
+          "default": "master",
+          "type": "string"
+        },
+        "integrationRepository": {
+          // The repository that hold generation branch, generation PR and integration branch.
+          // By default it's the same as mainRepository
+          "$ref": "#/definitions/RepositoryName"
+        },
+        "secondaryRepository": {
+          // Codegen runs on this repository.
+          // By default it's the same as 'mainRepository' but it could be different.
+          "$ref": "#/definitions/RepositoryName"
+        },
+        "secondaryBranch": {
+          // Codegen runs on this branch on secondaryRepository.
+          // By default it's the same as 'mainBranch' but it could be different.
+          "type": "string"
+        },
+        "integrationBranchPrefix": {
+          // The prefix that will be applied to the beginning of integration branches
+          "type": "string",
+          "default": "sdkAutomation"
+        },
+        "configFilePath": {
+          // Path to swagger-to-sdk config in sdk repo
+          "default": "swagger_to_sdk_config.json"
+        }
+      },
+      "required": [
+        "mainRepository"
+      ]
+    }
+  }
+}
+```
+
+### SwaggerToSdkConfig
+This is type of file `./swagger_to_sdk_config.json` in sdk repo.
+The running environment of these scripts would be expected to be __Ubuntu 18.04__ on Azure Pipeline. All the running script should be executable.
```

Comment:
I think it is fine to call out that linux is needed but we should be careful about being specific to Ubuntu 18.04. While that might be what we use today I expect that will update to newer version later.

In the future I think we will likely want to add some sort of advanced option that would allow languages to specify a docker container they can use to run the steps in instead of using the init script, but hopefully we don't need that initially.

Reply:
Makes sense. Currently let's make it simple.

Reply:
If you really want to use docker image, you can pull it in init and run everything in the image with folder mount I think.